analytics(trace-eap-waterfall): Centralizing trace state analytics

Abdkhan14 opened the pull request:
This PR centralizes the trace state analytics:
- Decoupling it from trace tree state 
- And waterfall renrenders

This is prep work for being able to 'scroll to and highlight nodes' on re-renders for trace to trace navigations (span links) and tab changes ( ai span waterfall to full trace waterfall), while ensuring that we don't send false analytics events.

Commit:
analytics(trace-eap-waterfall): Centralizing race state analytics

Commit:
analytics(trace-eap-waterfall): cleanup

gggritso commented on static/app/views/performance/newTraceDetails/useTraceStateAnalytics.tsx:
```diff
@@ -0,0 +1,101 @@
+        : 'unknown';
+      const issuesCount = TraceTree.UniqueIssues(traceNode).length;
+
+      traceAnalytics.trackTraceShape(
+        tree,
+        projects,
+        organization,
+        hasExceededPerformanceUsageLimit,
+        traceWaterfallSource,
+        traceAge,
+        issuesCount,
+        tree.eap_spans_count
+      );
```
👀 it looks like the only place that tracked this in the past is `TraceWaterfall`, and this PR will add it to `IssuesTraceWaterfall`, is that intentional? Also asking since the shape doesn't look like a state (unlike empty state and error state) so I'm wondering if it should be separate?

gggritso commented on static/app/views/performance/newTraceDetails/useTraceStateAnalytics.tsx:
```diff
@@ -0,0 +1,101 @@
+    trace.status,
+    trace.data,
+    trace.error,
+    meta,
```
This is a little scary, does `meta` change render-to-render? Just checking that this won't cause the analytics to run too often. Seems like it'd be safer to depend on just the span count, the only thing you need out of the meta.

gggritso reviewed (commented):
One thing that seems strange is that `AnrRootCause` seems to make a trace tree, but doesn't have analytics, maybe make sense to add something there? Otherwise just a few questions, mostly just from me not knowing what's going on

Commit:
fix(discover-event-details): Iterating on PR suggesstions

Commit:
feat(trace-eap-waterfall): Addressing PR suggestions

Commit:
feat(trace-eap-waterfall): Fixing tests

gggritso reviewed (approved):
👀 👍🏻

Abdkhan14 committed:
feat(trace-eap-waterfall): Relocating tracewaterfall models instantiation (#98604)

This is prep work for trace to trace navigations and preserving trace
view scrolled/highlighted row state on re-renders (for example: from
tabs switches)

We want the models responsible for rendering, to be instantiated on
load. The PR makes sure that this happens at the trace waterfall layer
and not a layer above anymore.

---------

Co-authored-by: Abdullah Khan <abdullahkhan@PG9Y57YDXQ.local>
Co-authored-by: getsantry[bot] <66042841+getsantry[bot]@users.noreply.github.com>

Abdkhan14 committed:
Merge branch 'master' into abdk/trace-analytics-uplift

Commit:
feat(trace-eap-waterfall): Resolving conflicts
